File: nodejs/src/client/workspace.ts
/**
 * Workspace facade (MCP: loxtep_workspace).
 * Delegates to projects, instances, and deployments APIs.
 * Version/snapshot REST is not yet available.
 */

import type { createProjectsApi } from './projects.js';
import type { createInstancesApi } from './instances.js';
import type { createDeploymentsApi } from './deployments.js';

export interface WorkspaceFacadeDeps {
  projects: ReturnType<typeof createProjectsApi>;
  instances: ReturnType<typeof createInstancesApi>;
  deployments: ReturnType<typeof createDeploymentsApi>;
}

/** Placeholder until version/snapshot REST endpoints ship (MCP-only today). */
export interface VersionsFacade {
  /** Not yet available over REST — use MCP list_versions / create_snapshot. */
  readonly unavailable: true;
}

export function createWorkspaceFacade(deps: WorkspaceFacadeDeps): {
  projects: WorkspaceFacadeDeps['projects'];
  instances: WorkspaceFacadeDeps['instances'];
  deployments: WorkspaceFacadeDeps['deployments'];
  versions: VersionsFacade;
} {
  return {
    projects: deps.projects,
    instances: deps.instances,
    deployments: deps.deployments,
    versions: { unavailable: true },
  };
}

export type WorkspaceFacade = ReturnType<typeof createWorkspaceFacade>;

File: python/src/loxtep/workspace.py
"""Workspace facade (MCP: loxtep_workspace)."""

from __future__ import annotations

from dataclasses import dataclass, field
from typing import Union

from .deployments import AsyncDeploymentsApi, DeploymentsApi
from .instances import AsyncInstancesApi, InstancesApi
from .projects import AsyncProjectsApi, ProjectsApi


@dataclass(frozen=True)
class VersionsFacade:
    unavailable: bool = True


@dataclass(frozen=True)
class WorkspaceFacade:
    projects: Union[ProjectsApi, AsyncProjectsApi]
    instances: Union[InstancesApi, AsyncInstancesApi]
    deployments: Union[DeploymentsApi, AsyncDeploymentsApi]
    versions: VersionsFacade = field(default_factory=VersionsFacade)
